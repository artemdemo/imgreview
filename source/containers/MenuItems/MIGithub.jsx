import React from 'react';
import Icon from '../../components/Icon/Icon';
import MainMenuItem from '../../components/MainMenu/MainMenuItem';

class MIGithub extends React.PureComponent {
    onClick = () => {
        window.location.href = 'https://github.com/artemdemo/img-review';
    };

    render() {
        return (
            <MainMenuItem
                onClick={this.onClick}
                right
            >
                <Icon
                    name='github'
                    title='Github'
                />
            </MainMenuItem>
        );
    }
}

export default MIGithub;
