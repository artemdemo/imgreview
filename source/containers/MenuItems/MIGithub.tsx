import React from 'react';
import Icon from '../../components/Icon/Icon';
import TopMenuItem from '../../components/TopMenu/TopMenuItem';

class MIGithub extends React.PureComponent {
    onClick = () => {
        window.location.assign('https://github.com/artemdemo/img-review');
    };

    render() {
        return (
            <TopMenuItem
                onClick={this.onClick}
            >
                <Icon
                    name='github'
                    title='Github'
                />
            </TopMenuItem>
        );
    }
}

export default MIGithub;
