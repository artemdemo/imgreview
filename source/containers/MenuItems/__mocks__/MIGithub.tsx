import React from 'react';

const MIGithub = props => (
    <div data-mock='MIGithub'>
        {props.children}
    </div>
);

export default MIGithub;
