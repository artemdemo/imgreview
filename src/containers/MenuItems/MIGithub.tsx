import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGithub } from '@fortawesome/free-brands-svg-icons';
import TopMenuItem from '../../components/TopMenu/TopMenuItem';

class MIGithub extends React.PureComponent {
    render() {
        return (
            <TopMenuItem
                href='https://github.com/artemdemo/img-review'
            >
                <FontAwesomeIcon icon={faGithub} />
            </TopMenuItem>
        );
    }
}

export default MIGithub;
