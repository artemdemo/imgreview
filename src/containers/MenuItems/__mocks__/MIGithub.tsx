import React from 'react';

const MIGithub = props => (
    <div data-mock='MIGithub'>
        {JSON.stringify(props, null, 2)}
    </div>
);

export default MIGithub;
