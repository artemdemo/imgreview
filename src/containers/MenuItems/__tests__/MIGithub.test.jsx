import React from 'react';
import renderer from 'react-test-renderer';
import MIGithub from '../MIGithub';

jest.mock('react-redux');
jest.mock('../../../components/TopMenu/TopMenuItem');

describe('MIGithub', () => {
    it('should render', () => {
        const tree = renderer.create(
            <MIGithub />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
