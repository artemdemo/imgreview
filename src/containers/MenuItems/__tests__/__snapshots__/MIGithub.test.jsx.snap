// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MIGithub should render 1`] = `
<div
  data-mock="TopMenuItem"
>
  <span
    data-mock="FontAwesomeIcon"
  >
    faGithub
  </span>
</div>
`;
